hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/TransactionMetadataBuilder.java

```diff
@@ -189,6 +218,20 @@ private boolean isNotNeeded(@NonNull final AccountID id) {
                 || payerKey == null;
     }
 
+    /**
+     * Checks if the metadata is already failed. In this case no need to look up that contract's key
+     * If the payer key has not been set we don't add other keys.
+     *
+     * @param id given contract
+     * @return true if the lookup is not needed, false otherwise
+     */
+    private boolean isNotNeeded(@NonNull final ContractID id) {
+        return id.equals(ContractID.getDefaultInstance())
```

Comment:
Maybe I'm nit-picking but I would do the null check for id here since this is the method that really does an access to it.

Reply:
The methods using this private method checks `Objects.requireNonNull` already.